cluster/gce/configure.sh

```diff
@@ -36,32 +38,143 @@ fetch_metadata() {
   fi
 }
 
-# DEPLOY_PATH is the gcs path where cri-containerd tarball is stored.
-DEPLOY_PATH=${DEPLOY_PATH:-"cri-containerd-release"}
+# fetch_env fetches environment variables from GCE metadata server
+# and generate a env file under ${CONTAINERD_HOME}. It assumes that
+# the environment variables in metadata are in yaml format.
+fetch_env() {
+  local -r env_name=$1
+  (
+    umask 700;
+    local -r tmp_env="/tmp/${env_name}.yaml"
+    tmp_env_content=$(fetch_metadata "${env_name}")
+    if [ -z "${tmp_env_content}" ]; then
+      echo "No environment variable is specified in ${env_name}"
+      return
+    fi
+    echo "${tmp_env_content}" > "${tmp_env}"
+    # Convert the yaml format file into a shell-style file.
+    eval $(python -c '''
+import pipes,sys,yaml
+for k,v in yaml.load(sys.stdin).iteritems():
+  print("readonly {var}={value}".format(var = k, value = pipes.quote(str(v))))
+''' < "${tmp_env}" > "${CONTAINERD_HOME}/${env_name}")
+    rm -f "${tmp_env}"
+  )
+}
+
+# is_preloaded checks whether containerd is preloaded in the image.
```

Comment:
nit: The function seems generic. Change to `whether a package has been preloaded`

Reply:
Done